src/clr-angular/utils/drag-and-drop/draggable.ts

```diff
@@ -3,40 +3,82 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import {AfterContentInit, Directive, ElementRef, OnDestroy} from "@angular/core";
+import {
+    AfterContentInit,
+    ComponentFactory,
+    ComponentFactoryResolver,
+    ContentChild,
+    Directive,
+    ElementRef,
+    EventEmitter,
+    Injector,
+    OnDestroy,
+    Output,
+    ViewContainerRef
+} from "@angular/core";
 import {Subscription} from "rxjs/Subscription";
+
+import {DomAdapter} from "../dom-adapter/dom-adapter";
+import {ClrDraggableGhost} from "./draggable-ghost";
+import {ClrIfDragged} from "./if-dragged";
 import {ClrDragEvent} from "./interfaces/drag-event";
 import {ClrDragEventListener} from "./providers/drag-event-listener";
 import {ClrDragHandleRegistrar} from "./providers/drag-handle-registrar";
+import {ClrDraggableStateRegistrar} from "./providers/draggable-state-registrar";
 
-
-@Directive(
-    {selector: "[clrDraggable]", providers: [ClrDragEventListener, ClrDragHandleRegistrar], host: {class: "draggable"}})
+@Directive({
+    selector: "[clrDraggable]",
+    providers: [ClrDragEventListener, ClrDragHandleRegistrar, ClrDraggableStateRegistrar, DomAdapter],
+    host: {class: "draggable"}
+})
 export class ClrDraggable<T> implements AfterContentInit, OnDestroy {
     private draggableEl: Node;
-
     private subscriptions: Subscription[] = [];
+    private componentFactory: ComponentFactory<ClrDraggableGhost<T>>;
 
     constructor(private el: ElementRef, private dragEventListener: ClrDragEventListener<T>,
-                private dragHandleRegistrar: ClrDragHandleRegistrar<T>) {
+                private dragHandleRegistrar: ClrDragHandleRegistrar<T>, private viewContainerRef: ViewContainerRef,
+                private cfr: ComponentFactoryResolver, private injector: Injector,
+                private draggableStateRegistrar: ClrDraggableStateRegistrar<T>) {
         this.draggableEl = this.el.nativeElement;
+        this.componentFactory = this.cfr.resolveComponentFactory<ClrDraggableGhost<T>>(ClrDraggableGhost);
     }
 
+    @ContentChild(ClrIfDragged) customGhost: ClrDraggableGhost<T>;
+
+    private createDefaultGhost(event: ClrDragEvent<T>) {
+        this.draggableStateRegistrar.register(this.draggableEl, event);
+        this.viewContainerRef.createComponent(this.componentFactory, 0, this.injector,
```

Comment:
Can you please just add a comment explaining that in this case, the ghost will be **outside** of the draggable in the DOM (which means a sibling)? I just want to have something written in case we forget and decide to base our styling based on parent selectors..

Reply:
Sure, will do that.

Reply:
Added the comment.